py/mistql/cli.py

```diff
@@ -34,20 +35,34 @@
 
 
 def main(supplied_args=None):
+    
     if supplied_args is None:
         args = parser.parse_args()
     else:
         args = parser.parse_args(supplied_args)
     raw_data: Union[str, bytes]
+    skip_json_load = False
     if args.data:
         raw_data = args.data
-    elif args.file:
-        with open(args.file, 'rb') as f:
-            raw_data = f.read()
+    elif args.file: 
+         
+        file_ext = args.file.split(".")[-1] 
+        if file_ext in ["jsonl"]: 
+            skip_json_load = True
+            out = []
+            with open(args.file, 'rb') as f:
+                for item in json_lines.reader(f):                    
+                    out.append( query(args.query, item) )
```

Comment:
this isn't necessary to be fixed before merge because it's a bigger change and will probably require some more restructuring than you might be willing to do, but this should really be streaming, e.g. output should be written as it is received, not when the input stream reaches eof.

Reply:
@evinism thanks, that was only the first version. 
I haven't started with streams since i had doubts about cases like using count and group by. 

any suggestions how to make a stream process?

Reply:
thanks for your patience! I was on vacation!

the key for this would be opening the output stream simultaneously, whether the output stream is stdout or a file, and reading a line from the json_lines reader, and outputting it, without storing the results intermediately. 

Also, it'd be highly beneficial to also extend the --file-jsonl to stdin as well, such that the piping command `tail -f ./logs/production.jsonl | python -m mistql.cli foo.bar` will work. We can certainly separate that out into its own PR, too, if you just want to get somethign merged.